### PR TITLE
hotfix# typescript build

### PR DESCRIPTION
**Pull request by Amar-NHS**
Fix for getting the FrontEnd stable on ElasticBeanstalk. Added deployment stage steps for nextauth.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Comment by avi312NHS** [comment]
good to see TS being introduced and prop-types being removed
- have you checked if the app can still build without any compilation error?
- are you able to unstage the console.logs so they don't go into main and you just keep them locally

File: src/app/components/NavMenu.jsx [hunk 1]
```diff
@@ -1,9 +1,18 @@
+"use client";
 import Link from "next/link";
 import "../styles/css/sass.css";
+import { useEffect, useState } from "react";
 import { signOut } from "next-auth/react";
 
 // Navigation Menu
 export default function NavMenu() {
+  const [currentUrl, setCurrentUrl] = useState("");
```
Comment (avi312NHS) [inline src/app/components/NavMenu.jsx:9]: This can be const [currentUrl, setCurrentUrl] = useState<string>("");

Comment (Amar-NHS) [reply src/app/components/NavMenu.jsx:9]: Thought type annotations only can be used in typescript file.

**Comment by Amar-NHS** [comment]
> good to see TS being introduced and prop-types being removed
> 
> * have you checked if the app can still build without any compilation error?
> * are you able to unstage the console.logs so they don't go into main and you just keep them locally


yeh, checked and builds fine. will get the console logs removed.

File: src/app/components/NavMenu.jsx (hunk 1 above)
Comment (avi312NHS) [reply src/app/components/NavMenu.jsx:9]: ahh thought this was a converted tsx file

**Comment by avi312NHS** [comment]
> > good to see TS being introduced and prop-types being removed
> > 
> > * have you checked if the app can still build without any compilation error?
> > * are you able to unstage the console.logs so they don't go into main and you just keep them locally
> 
> yeh, checked and builds fine. will get the console logs removed.

so `npm run build` works fine?

**Comment by Amar-NHS** [comment]
> > > good to see TS being introduced and prop-types being removed
> > > 
> > > * have you checked if the app can still build without any compilation error?
> > > * are you able to unstage the console.logs so they don't go into main and you just keep them locally
> > 
> > 
> > yeh, checked and builds fine. will get the console logs removed.
> 
> so `npm run build` works fine?

yes, did that with Andrew to check if it builds fine.

**Review by avi312NHS — APPROVED** [review]

File: src/app/views/invitation_planning/NationalForecastUptakeTable.jsx
Comment (zainm97) [inline src/app/views/invitation_planning/NationalForecastUptakeTable.jsx:1]: prop type left in

File: src/app/views/invitation_planning/QuintileTargetTable.jsx
Comment (zainm97) [inline src/app/views/invitation_planning/QuintileTargetTable.jsx:1]: prop type left in

File: src/app/views/invitation_planning/QuintileTargetTable.test.js
Comment (zainm97) [inline src/app/views/invitation_planning/QuintileTargetTable.test.js:1]: do we need to remove?

File: src/app/views/invitation_summary/SummaryListFirst.jsx
Comment (zainm97) [inline src/app/views/invitation_summary/SummaryListFirst.jsx:1]: prop type left in

File: src/app/views/invitation_summary/SummaryListSecond.jsx
Comment (zainm97) [inline src/app/views/invitation_summary/SummaryListSecond.jsx:1]: prop type left in

File: src/app/views/privacy_confirmation/PrivacyConfirmation.test.js [hunk 2]
```diff
@@ -1,33 +1,80 @@
...
+      </SessionProvider>
+    );
     expect(screen.getByText("Protecting patient data")).toBeInTheDocument();
-    expect(screen.getByText("You must only access a participant’s records if there is a legitimate requirement to access their data.")).toBeInTheDocument();
-    expect(screen.getByText("All activity in the Galleri pilot system is routinely checked to ensure records are not accessed inappropriately.")).toBeInTheDocument();
-    expect(screen.getByText("When you are finished using the system you must:")).toBeInTheDocument();
-    expect(screen.getByText("Log out using the button in the top right hand corner.")).toBeInTheDocument();
-    expect(screen.getByText("Remove your smart card.")).toBeInTheDocument();
-    expect(screen.getByText("I have read and understood this message")).toBeInTheDocument();
+    // expect(screen.getByText("You must only access a participant's records if there is a legitimate requirement to access their data.")).toBeInTheDocument();
```
Comment (zainm97) [inline src/app/views/privacy_confirmation/PrivacyConfirmation.test.js:38]: can remove?

**Review by zainm97 — DISMISSED** [review]
looks good. Couple comments left in but apart from that logic is fine

File: src/app/views/invitation_planning/ErrorInvitations.jsx
[the remark this answers is not in the document]
Comment (Amar-NHS) [reply src/app/views/invitation_planning/ErrorInvitations.jsx:1]: will clean up on my next PR.

File: src/app/views/invitation_planning/QuintileTargetTable.test.js
Comment (Amar-NHS) [reply src/app/views/invitation_planning/QuintileTargetTable.test.js:1]: will clean up on my next PR.

File: src/app/views/invitation_summary/SummaryListSecond.jsx
Comment (Amar-NHS) [reply src/app/views/invitation_summary/SummaryListSecond.jsx:1]: will clean up on my next PR.

File: src/app/views/privacy_confirmation/PrivacyConfirmation.test.js (hunk 2 above)
Comment (Amar-NHS) [reply src/app/views/privacy_confirmation/PrivacyConfirmation.test.js:38]: thanks for pointing it out. will remove it on next PR.

File: src/app/views/invitation_summary/SummaryListFirst.jsx
Comment (Amar-NHS) [reply src/app/views/invitation_summary/SummaryListFirst.jsx:1]: will clean up on my next PR.

File: src/app/views/invitation_planning/QuintileTargetTable.jsx
Comment (Amar-NHS) [reply src/app/views/invitation_planning/QuintileTargetTable.jsx:1]: will clean up on my next PR.

**Review by zainm97 — APPROVED** [review]
look good

**Pull request merged**